src/main/kotlin/ch/hevs/cloudio/cloud/restapi/endpoint/permission/EndpointPermissionController.kt

```diff
@@ -47,13 +47,9 @@ class EndpointPermissionController(
         }.id.let {userID ->
             when {
                 permission == EndpointPermission.DENY -> userEndpointPermissionRepository.deleteByUserIDAndEndpointUUID(userID, uuid)
-                permission.fulfills(EndpointPermission.WRITE) -> {
-                    val endpointPermission = userEndpointPermissionRepository.findByUserIDAndEndpointUUID(userID, uuid).orElse(UserEndpointPermission(userID, uuid, permission))
-                    endpointPermission.modelPermissions.clear()
-                    userEndpointPermissionRepository.save(endpointPermission)
-                }
                 else -> {
                     val endpointPermission = userEndpointPermissionRepository.findByUserIDAndEndpointUUID(userID, uuid).orElse(UserEndpointPermission(userID, uuid, permission))
+                    endpointPermission.permission = permission
```

Comment:
We should still call `endpointPermission.modelPermissions.clear()` to get rid of model permission as they are not longer needed.